### PR TITLE
Add Users authentication and authorisation

### PR DESCRIPTION
**Pull request by nedkamburov**
- Add gems like Devise and Pundit
- Add User model
- Add roles for the User (user and admin) 
- Generate migrations
- Add several random users to in the seeds + one admin user by default 
- Add Pages controller and view for the home page
- Add an Admin Namespace with Pundit so only an admin can access /admin pages

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+  validates :name, presence: true
+
+  enum role: [:user, :admin]
+  after_initialize :set_default_role, :if => :new_record?
+
+  def set_default_role
+    self.role ||= :user
+  end
+  def is_admin?
```
Comment (igor04) [inline app/models/user.rb:15]: with ruby it's more convenient to have `admin?` without `is_..`, also `enum` already defined such method for you, or maybe `role_admin?`

Comment (nedkamburov) [reply app/models/user.rb:15]: Thanks Igor, I just checked and current_user does indeed have admin? method. I am fairly sure I was looking for that but something didn't work out. Maybe I misspelled the variable current_user somewhere. 
Anyways, it's all good now.

File: Gemfile [hunk 2]
```diff
@@ -70,3 +70,11 @@ group :test do
   gem "selenium-webdriver"
   gem "webdrivers"
 end
+
+gem "devise", "~> 4.8"
+
+gem "faker", "~> 2.23"
```
Comment (igor04) [inline Gemfile:76]: it need to be part of `test` group

File: Gemfile [hunk 3]
```diff
@@ -70,3 +70,11 @@ group :test do
...
+
+gem "pundit", "~> 2.2"
+
+gem "pry", "~> 0.14.1"
```
Comment (igor04) [inline Gemfile:80]: `test` and `development` group

File: app/models/user.rb [hunk 4]
```diff
@@ -0,0 +1,23 @@
+class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
+
+  validates :name, presence: true
+
+  enum role: [:user, :admin]
+  after_initialize :set_default_role, :if => :new_record?
```
Comment (igor04) [inline app/models/user.rb:10]: new syntax `if: :new_record?` would be better

**Review by igor04 — APPROVED** [review]

**Pull request merged**